ui/components/failed.jsx

```diff
@@ -0,0 +1,173 @@
+import _ from 'lodash';
+import React from 'react';
+import ImageDiff from 'react-image-diff';
+import classNames from 'classnames';
+import {Clearfix, ButtonGroup, Button,
+  Jumbotron, Grid, Row, Col, Panel}
+  from 'react-bootstrap';
+
+function _renderDefaultView(paths) {
+  return <div className="simple">
+    <img className="diff"
+         src={paths.diff}
+         key={paths.diff} />
+  </div>;
+}
+
+function _render2UpView(paths) {
+  return <div className="simple">
+    <Grid>
+      <Row className="show-grid">
+        <Col xs={12} md={6}>
+          <img className="baseline"
+               src={paths.baseline}
+               key={paths.baseline} />
+        </Col>
+        <Col xs={12} md={6}>
+          <img className="screenshot"
+               src={paths.screenshot}
+               key={paths.screenshot} />
+        </Col>
+      </Row>
+    </Grid>
+  </div>;
+}
+
+function _renderSwipeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="swipe"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _renderFadeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="fade"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _getSelectViewButtons(currentView, viewOptions, onViewChange) {
+  let buttons = [];
+
+  viewOptions.forEach(function(item) {
+    buttons.push(
+      <Button name={item}
+              key={item}
+              onClick={onViewChange}
+              className={classNames({active: item === currentView})}>
+        {item}
+      </Button>
+    );
+  });
+
+  return buttons;
+}
+
+class FailedTest extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      view: 'default',
+      value: 0.5,
+      openError: false
+    };
+
+    this.onValueChange = this.onValueChange.bind(this);
+    this.onViewChange = this.onViewChange.bind(this);
+    this.onErrorMessageOpen = this.onErrorMessageOpen.bind(this);
+  }
+
+  render() {
+    const {paths, error} = this.props;
+
+    return <div className="diffs">
+      <Button bsStyle="danger"
+              bsSize="xsmall"
+              onClick={this.onErrorMessageOpen}>
+        Show Error
+      </Button>
+
+      <Panel collapsible expanded={this.state.openError} bsStyle="danger">
+        {_.isUndefined(paths)
+          ? <p> This test did not fail because Mugshot found differences :( </p>
```

Comment:
So this is to catch errors that are thrown before the diffing? Like WebdriverIO errors? We should display the actual error in that case.